cms/djangoapps/contentstore/views/item.py

```diff
@@ -1135,10 +1134,12 @@ def create_xblock_info(xblock, data=None, metadata=None, include_ancestor_info=F
     if is_concise:
         if child_info and len(child_info.get('children', [])) > 0:
             xblock_info['child_info'] = child_info
+        user_partitions = get_user_partition_info(xblock, schemes=['random'], course=course)
```

Comment:
OK, I understand now where user_partitions is used.

I think the clearest thing is to not pass user_partitions into get_group_display_name, and instead have that method fetch the user_partitions it cares about (the random scheme ones).

I'd also recommend making the method name and doc clear that it is dealing with groups within the random UserPartition only. All UserPartitions have groups. Perhaps "get_random_group_display_name".